tv_apps/fling-player/js/fling_player.js

```diff
@@ -413,9 +446,26 @@
   proto.onLoadRequest = function (e) {
     this.resetUI();
     this.showLoading(true);
-    this.toggleInitialMessage(true);
     this._player.load(e.url);
     this.play();
+    this._connector.getControllingDeviceInfo()
+                   .then((info) => {
+                      var msg = this._initialMsg.textContent;
+                      var name = 'device';
+                      if (info.displayName &&
+                          typeof info.displayName == 'string'
+                      ) {
+                        name = info.displayName;
+                      }
+
+                      this._initialMsg.textContent =
+                          msg.replace('{{ device }}', name);
+
+                      this.toggleInitialMessage(true);
+                      if (this._player.isPlaying()) {
+                        this.toggleInitialMessage(false); // hide later
```

Comment:
Add the opt paramter to indicate that hidding the initail message is delayed for better and clearer meaning here, thanks.